mkosi/kmod.py

```diff
@@ -235,82 +235,78 @@ def resolve_module_dependencies(
         builtin = set(module_path_to_name(Path(m)) for m in p.read_text().splitlines())
     else:
         builtin = set()
-    with chdir(context.root):
-        allmodules = set(modulesd.rglob("*.ko*"))
-    nametofile = {module_path_to_name(m): m for m in allmodules}
-
-    log_step("Running modinfo to fetch kernel module dependencies")
 
-    # We could run modinfo once for each module but that's slow. Luckily we can pass multiple modules to
-    # modinfo and it'll process them all in a single go. We get the modinfo for all modules to build two maps
-    # that map the path of the module to its module dependencies and its firmware dependencies
-    # respectively. Because there's more kernel modules than the max number of accepted CLI arguments, we
-    # split the modules list up into chunks.
-    info = ""
-    for i in range(0, len(nametofile.keys()), 8500):
-        chunk = list(nametofile.keys())[i : i + 8500]
-        info += modinfo(context, kver, chunk)
-
-    log_step("Calculating required kernel modules and firmware")
+    log_step("Resolving dependencies for kernel modules and firmware")
 
     moddep = {}
     firmwaredep = {}
 
     depends: set[str] = set()
     firmware: set[Path] = set()
 
-    with chdir(context.root):
-        for line in info.split("\0"):
-            key, sep, value = line.partition(":")
-            if not sep:
-                key, sep, value = line.partition("=")
+    todo = list(modules) + list(builtin)
+    done = set()
```

Comment:
Not sure why we have to maintain `done`? Just subtract `chunk` from todo every step and you'll eventually end up empty.